packages/shortest/src/ai/client-provider.ts

```diff
@@ -0,0 +1,140 @@
+import assert from "node:assert";
+import { createAnthropic } from "@ai-sdk/anthropic";
+import { createAmazonBedrock } from "@ai-sdk/amazon-bedrock";
+import { LanguageModelV1 } from "ai";
+import { LLMConfig } from "../types/ai";
+
+/**
+ * Creates an AI client based on the provided configuration.
+ *
+ * @param config - The configuration for the language model.
+ * @returns A LanguageModelV1 instance.
+ * @throws Will throw an error if the provider or model is unsupported or if required environment variables are missing.
+ */
+export function createAIClient(config: LLMConfig): LanguageModelV1 {
+  const { provider, region, accessKeyId, secretAccessKey } = config;
+  const model = resolveWhitelistedProviderModel(provider, config.model);
+
+  switch (provider) {
+    case "anthropic":
+      return createAnthropicClient(model);
+
+    case "amazon-bedrock":
```

Comment:
Should keep this PR atomic, focused on Anthropic. AmazonBedrock can be implemented in a follow-up PR, see https://github.com/anti-work/shortest/issues/310